src/map/entities/petentity.cpp

```diff
@@ -206,12 +237,58 @@ void CPetEntity::Spawn()
         mobutils::GetAvailableSpells(this);
     }
 
+    if (m_PetType == PET_TYPE::JUG_PET)
+    {
+        m_jugSpawnTime = server_clock::now();
+    }
+
     // NOTE: This is purposefully calling CBattleEntity's impl.
     // TODO: Calling a grand-parent's impl. of an overridden function is bad
     CBattleEntity::Spawn();
     luautils::OnMobSpawn(this);
 }
 
+bool CPetEntity::shouldPersistThroughZone()
+{
+    // TODO: Add other types of pets that should persist through zone here
+    return getPetType() == PET_TYPE::WYVERN ||
+           getPetType() == PET_TYPE::AUTOMATON ||
+           (getPetType() == PET_TYPE::JUG_PET && settings::get<bool>("map.KEEP_JUGPET_THROUGH_ZONING"));
+}
+
+bool CPetEntity::shouldDespawn(time_point tick)
+{
+    if (isCharmed && tick > charmTime)
```

Comment:
isCharmed should be impossible on a CPetEntity, this is effectively dead code. Mobs change allegiance to players in order to function like pets when charmed. Additionally, pets are immune to charm directly and masters who get charmed immediately have their pets despawned

Reply:
This is just moving the existing code that was outside of "shouldDespawn".
So it's safe to just remove this?

Reply:
ah well.. if it was there, it would be better to add a comment to figure it out and clean it up later. That's outside the scope of your PR.

Reply:
Done

> For commit messages: think about the actual work you're trying to achieve and what might be meaningful to people looking at your commits in the future, and not the work you're doing to try and get the PR to pass CI.
> 
> * Addressing things from review
> * Formatting code
> * Oops, forgot a semicolon!
> * lol
> 
> etc. All have nothing to do with the actual code and logic changes you're trying to do.
> 
> Hope that gives some guidance 👍

Ya that makes sense. I've been pretty spoiled by squashing commits when merging PRs in my current team. Will make sure to fixup commits before merge. It does sound like a task that should be done after the PR has been approved though. Otherwise it may be too much friction. Thoughts?